detox/src/devices/drivers/android/tools/FreeDeviceFinderBase.test.js

```diff
@@ -0,0 +1,87 @@
+const FreeDeviceFinderBase = require('./FreeDeviceFinderBase');
+
+describe('devices lookup helper', () => {
+  const mockAdb = { devices: jest.fn() };
+
+  let mockDeviceRegistry;
+  let mockIsDeviceMatching;
+  let uut;
+  beforeEach(() => {
+    mockDeviceRegistry = { isDeviceBusy: jest.fn() };
+    mockIsDeviceMatching = jest.fn().mockResolvedValue(false);
+
+    const FreeDeviceFinderImpl = class extends FreeDeviceFinderBase {
+      constructor(adb, deviceRegistry) {
+        super(adb, deviceRegistry);
+        this.isDeviceMatching = mockIsDeviceMatching;
+      }
+    }
+    uut = new FreeDeviceFinderImpl(mockAdb, mockDeviceRegistry);
+  });
+
+  it('should return null when there are no devices', async () => {
+    mockNoAdbDevices();
```

Comment:
Can't we just use `mockAdbDevices([]);`?

Reply:
Changed